lms/djangoapps/student_account/views.py

```diff
@@ -171,6 +172,8 @@ def password_change_request_handler(request):
     if email:
         try:
             request_password_change(email, request.get_host(), request.is_secure())
+            user = User.objects.get(email=email) if not user.is_authenticated() else user
```

Comment:
This will read a bit more clearly if you reverse the logic:

```python
user = user if user.is_authenticated() else User.objects.get(email=email)
```